I18n

biwano opened the pull request:
Integrating i18n
You have to `yarn install` && `yarn lingui:compile` to setup translations

Commit:
Added Sublime Text Files to .gitignore

Commit:
Added a message framework and a console interceptor.

Commit:
MessagesSlice written in typescript
Using SnackBar component for messages

Commit:
Using useDispatch() instead of store.dispatch where possible
Removed "set" action from MessageStore

Commit:
Fixed comments typos

Commit:
Typascript fixes in Bondslice.ts

Commit:
Metamask setup in e2e tests

Commit:
Stake test complete

Commit:
Updated README and comments

Commit:
Merge branch 'develop' of https://github.com/OlympusDAO/olympusdao into e2e_unit_tests

goatsOnDrugs committed:
Merge branch 'e2e_unit_tests' of https://github.com/biwano/olympus-frontend into develop

goatsOnDrugs committed:
adding assertions to e2e stake and comments to stake int

goatsOnDrugs committed:
wip

goatsOnDrugs committed:
Merge branch 'develop' of https://github.com/OlympusDAO/olympusdao into test/227d

Commit:
Added Frax bond tests (bond, claim, claim & stake) (#470)

Added helpers.js to hold reusable functions

goatsOnDrugs committed:
Merge branch 'test/227d' of https://github.com/OlympusDAO/olympusdao into test/227d

Commit:
.env reader for synpress (#480)

* Added Frax bond tests (bond, claim, claim & stake)
Added helpers.js to hold reusable functions

* Added .env reader for synpress

goatsOnDrugs committed:
Merge branch 'test/227d' of https://github.com/OlympusDAO/olympusdao into test/227d

goatsOnDrugs committed:
added check for windows machine on test launcher

Commit:
Translation setup

Commit:
Added locale switcher button

Commit:
Added translations to sidebar and top menu

Commit:
Fixed connectbutton compilation issue

Commit:
Added some top menu translations

Commit:
Translated bond pages

Commit:
Merge branch 'develop' into i18n

Commit:
Added language files

Commit:
Merge branch 'develop' into i18n

goatsOnDrugs committed:
wip

Commit:
Translated 3,3 Together

Commit:
Merge branch 'develop' into i18n

Commit:
Translated stake page

Commit:
Translated dashboards

Commit:
English is the default language

Commit:
Merge branch 'develop' into i18n

Zayen-X commented on .gitignore:
```diff
@@ -40,3 +40,6 @@ yarn-error.log*
 *.sublime-*
+
+# Compiled locales
+#src/locales/*/*.js
```
why do we not want to use compiled locales?

biwano replied:
I did not want to put the compiled local in the repo. I think they should be compiled as part of the build process. But i am open to advice.

biwano replied:
For the Trans object and t method. I think they are mandatory. They enable languijs cli to find the messages to be translated. If they were no markers langui would not know how to prepare the files.

wollemipine replied:
not sure how others see this, but the `t` method has been pretty standard amongst most projects i have seen that have implement internationalization...is clunky though, agree :-)

Zayen-X commented on src/components/Sidebar/NavContent.jsx:
```diff
@@ -1,12 +1,13 @@
 import { NavLink } from "react-router-dom";
 import Social from "./Social";
-import externalUrls from "./externalUrls";
+import externalUrls from "./externalUrls.jsx";
```
Don't include  extensions here.

biwano replied:
Will do!

Zayen-X reviewed (dismissed):
The setup looks reasonable. I have a general question on usage. 

Is there anyway we can implement this without explicitly calling the `Trans` object, or the `t` method? Feels cumbersome.

goatsOnDrugs committed:
pulled dev and working through bugs

goatsOnDrugs committed:
Merge branch 'develop' of https://github.com/OlympusDAO/olympusdao into test/227d

goatsOnDrugs committed:
integration test are passing now

0xMakka committed:
Dynamic imports for translations

Commit:
Merge branch 'develop' into i18n

Commit:
fixed jsx extension in externalUrls import

Commit:
Changed transloations folder in order to prepare use of a submodule

Commit:
Merge branch 'develop' into i18n

Commit:
Translation submodule setup

Commit:
Merge branch 'develop' of https://github.com/OlympusDAO/olympus-frontend into develop

Commit:
Merge branch 'test/227d' of https://github.com/OlympusDAO/olympus-frontend into develop

Commit:
A little bit for refactoring

Commit:
More refactoring

Commit:
A little bit of refactoring

Commit:
Refactorisation
Added Korean language

Commit:
Merge remote-tracking branch 'origin' into i18n

biwano commented:
Hi guys. I think that PR is ready from a code point of view.
For logistics:
* How/When do we pass the information about the usage of Trans segment and t function ?
* Should we put the compilation of locale files into the build process or do we keep that manual ?
* How do we inform the translation team when we need new stuff to be translated?

It would be nice if we could merge that even with translations disabled because it is breaking every time someone commits something into the develop branch

Commit:
Merge branch 'develop' into i18n

Commit:
Removed useless console.log

Commit:
Merge branch 'develop' into i18n

Commit:
Added turkish language
Modified odds display to accomodate potential translation

Commit:
make "Buy on" translations use macro

Commit:
Merge branch 'develop' into i18n

Commit:
Merge branch 'develop' into HEAD

Commit:
Fixed yarn.lock

Commit:
Merge branch 'i18n' of https://github.com/biwano/olympus-frontend into HEAD

Commit:
Merge branch 'develop' into i18n

Commit:
Fixed package.json

Commit:
Merge branch 'develop' into i18n

appleseed-iii commented:
action items to discuss with @biwano:

- [ ] How/When do we pass the information about the usage of Trans segment and t function ?
- [x] Should we put the compilation of locale files into the build process or do we keep that manual ?
- [ ] How do we inform the translation team when we need new stuff to be translated?

appleseed-iii commented on README.md:
```diff
@@ -34,25 +36,46 @@ Open the source code and start editing!
+- Actually run the tests
+  `yarn synpress:run`
+
+
```
is this applicable to this PR? Or should it be removed until @JoeyR84 's PR for testing?
Or should we make it more clear right now that tests only apply to translations at this time?

biwano replied:
No Sry, Maybe a bad merge.
Removed.

appleseed-iii commented on README.md:
```diff
@@ -63,46 +86,51 @@ The files/folder structure are a  **WIP** and may contain some unused files. The
+The language files are located in `src/locales`
+
+- Use `yarn lingui:extract` to scan the code for new translatable entries add them to the po files
+- Use `yarn lingui:compile` to compile the po files into the javascript files actually used by the application
```
add a note about contacting the Discord Translations channel when you have a new feature that needs translation.

appleseed-iii replied:
also add a note telling devs to encapsulate strings that should be translated in `<Trans>`

biwano replied:
Done. Do not hesitate to tell me if my english is bad. I won't take it personaly.

appleseed-iii commented on README.md:
```diff
@@ -63,46 +86,51 @@ The files/folder structure are a  **WIP** and may contain some unused files. The
-Auto deployed on [Fleek.co](http://fleek.co/) fronted by [Cloudflare](https://www.cloudflare.com/). Since it is hosted via IPFS there is no running "server" component and we don't have server sided business logic. Users are served an `index.html` and javascript to run our applications. 
 
-_**TODO**: TheGraph implementation/how/why we use it._
+Auto deployed on [Fleek.co](http://fleek.co/) fronted by [Cloudflare](https://www.cloudflare.com/). Since it is hosted via IPFS there is no running "server" component and we don't have server sided business logic. Users are served an `index.html` and javascript to run our applications.
```
appleseed to update the build process for lingui

appleseed-iii commented on src/components/TopBar/LocaleSwitch.tsx:
```diff
@@ -0,0 +1,68 @@
+  return locales[locale].flag;
+}
+
+function LocaleSwitcher() {
```
- [x] appleseed to check this on mobile

appleseed-iii commented on src/lib/ohmToast.js:
```diff
@@ -8,7 +7,7 @@ function Alert(props) {
 }
 
-const useStyles = makeStyles(theme => ({
+useStyles = makeStyles(theme => ({
```
why removing `const`?

appleseed-iii reviewed (commented):
some initial commentary based on discussions with @biwano . Will be loading this PR locally later today.

Commit:
Added some explanations about the translation workflow in the README

Commit:
Merge branch 'develop' into i18n

appleseed-iii commented:
- [x] is language setting stored in browser & saved on refreshes

Commit:
Merge branch 'develop' into i18n

Commit:
Updated README with instructions to:
- use linguijs t and Trans idioms
- use the app-translations discord channel to inform translators

Commit:
Corrected bad merge with test/227d

biwano commented:
I realized I had the test/227d branch merged in this PR. Really sorry about that I did not check my PR correctly. I fixed it now.

biwano commented:
> * [x]  is language setting stored in browser & saved on refreshes

Not yet. rms rob proposed to do something with graph storage. So I am waiting for that. I can implement it with HTML locale storage for the time being if you want.

appleseed-iii commented:
> > * [x]   is language setting stored in browser & saved on refreshes
> 
> Not yet. rms rob proposed to do something with graph storage. So I am waiting for that. I can implement it with HTML locale storage for the time being if you want.

Yea I think implement LocalStorage for now so that we have something. Then when rob is done we can switch over to his state

Commit:
Saving user prefered locale into localstorage (like theme)

biwano commented:
> > > * [x]    is language setting stored in browser & saved on refreshes
> > 
> > 
> > Not yet. rms rob proposed to do something with graph storage. So I am waiting for that. I can implement it with HTML locale storage for the time being if you want.
> 
> Yea I think implement LocalStorage for now so that we have something. Then when rob is done we can switch over to his state

Done. I copied what has been done for theme. I think I will propose in another PR to create a slice for user preferences.

Commit:
Merge branch 'develop' into i18n

Commit:
Removed e2e artifact in package.json

Commit:
Fixed package.json

Commit:
Merge branch 'develop' into i18n

Commit:
phase 1 done

Commit:
Merge pull request #20 from OlympusDAO/ci-add-translations

phase 1 done

Commit:
Merge branch 'develop' into i18n

Commit:
Merge branch 'i18n' of https://github.com/biwano/olympus-frontend into i18n

Commit:
Merge branch 'develop' into i18n

appleseed-iii commented on src/views/33Together/33Together.jsx:
```diff
@@ -143,6 +144,42 @@ const PoolTogether = () => {
+          />
+        </TabPanel>
+      </Paper>
+=======
```
this merge can be resolved by copying lines 198 & 199 & pasting between lines 161-162

appleseed-iii commented on package.json:
```diff
@@ -52,6 +56,7 @@
     "isomorphic-fetch": "^3.0.0",
     "lodash": "^4.17.21",
     "material-ui": "^0.20.2",
+    "msw": "^0.35.0",
```
mock service worker

appleseed-iii commented on package.json:
```diff
@@ -85,6 +93,8 @@
     "@typescript-eslint/parser": "^4.33.0",
     "autoprefixer": "^10.2.4",
+    "babel-plugin-lodash": "^3.3.4",
+    "babel-plugin-transform-imports": "^2.0.0",
```
why are these two necessary?

biwano replied:
If I dont have that in node_modules, `yarn lingui:extract` gives warnings.

appleseed-iii commented on package.json:
```diff
@@ -123,6 +135,8 @@
     "watch": "node ./scripts/watch.js",
+    "lingui:extract": "lingui extract",
+    "lingui:compile": "lingui compile",
     "postinstall": "yarn run typechain --target ethers-v5 --out-dir src/typechain src/abi/*.json src/abi/**/*.json"
```
what if we place lingui:compile as a `postinstall` script since devs need it too?

appleseed-iii reviewed (commented):
What is the proper way to get the translations onto my dev machine? When I switch flags I still only see english.

nvm:

```
yarn build
```

but this wouldn't be necessary in most cases.

appleseed-iii commented on src/views/Stake/Stake.jsx:
```diff
@@ -259,10 +272,9 @@ function Stake() {
                     >
-                      <Tab label="Stake" {...a11yProps(0)} />
-                      <Tab label="Unstake" {...a11yProps(1)} />
+                      <Tab label={t`do_stake`} {...a11yProps(0)} />
```
should this be
```
t`Stake`
```

biwano replied:
I had to differentiate "Stake" the noun and "Stake" the verb because in many language they are not translated the same. Same for bonds.

appleseed-iii replied:
ok, interesting

appleseed-iii commented on src/views/Stake/Stake.jsx:
```diff
@@ -371,15 +383,25 @@ function Stake() {
+                      </Typography>
+                      <Typography variant="body1">
+                        <Trans>Your Balance</Trans>
+                      </Typography>
```
duplicate of line 386-388 above

appleseed-iii commented on src/views/Stake/Stake.jsx:
```diff
@@ -371,15 +383,25 @@ function Stake() {
+                      </Typography>
+                      <Typography variant="body1">
+                        <Trans>Your Staked Balance</Trans>
+                      </Typography>
```
duplicate of line 398-400 above

appleseed-iii commented on src/views/Stake/Stake.jsx:
```diff
@@ -371,15 +383,25 @@ function Stake() {
+                      <Typography variant="body1" id="user-staked-balance">
                         {isAppLoading ? <Skeleton width="80px" /> : <>{trimmedBalance} sOHM</>}
                       </Typography>
                     </div>
```
lines 411, 414, 420, 423 (unedited below this comment) need translations. What is the proper way for a dev to notify you?

biwano replied:
The process would be the following:
- The dev includes t and Trans into their code
- When their code is merged with dev. CI executes lingui:extract and stores the result somewhere (JesusfChrist is evaluating if this is possible)
- A discord bot posts the result in the app-translation discord chat.

appleseed-iii commented on src/views/Bond/Bond.jsx:
```diff
@@ -86,8 +87,8 @@ function Bond({ bond }) {
               >
-                <Tab label="Bond" {...a11yProps(0)} />
-                <Tab label="Redeem" {...a11yProps(1)} />
+                <Tab aria-label="bond-tab-button" label={t`do_bond`} {...a11yProps(0)} />
```
should this be bond rather than do_bond

biwano replied:
Same as Stake above

appleseed-iii committed:
fixing merge commit & a couple other ui bugs

appleseed-iii commented on src/views/33Together/33Together.jsx:
```diff
@@ -144,7 +144,6 @@ const PoolTogether = () => {
   return (
     <div id="pool-together-view">
       <PoolPrize />
-<<<<<<< HEAD
```
merge commit fix

appleseed-iii commented on src/Root.tsx:
```diff
@@ -1,19 +1,29 @@
-        <BrowserRouter basename={"/#"}>
-          <App />
-        </BrowserRouter>
+        <I18nProvider i18n={i18n}>
+          <BrowserRouter basename={"/#"}>
+            <App />
+          </BrowserRouter>
+        </I18nProvider>
```
merge commit fix

appleseed-iii commented on package.json:
```diff
@@ -137,7 +137,8 @@
     "lingui:compile": "lingui compile",
-    "postinstall": "yarn run typechain --target ethers-v5 --out-dir src/typechain src/abi/*.json src/abi/**/*.json"
+    "typechain:build": "yarn run typechain --target ethers-v5 --out-dir src/typechain src/abi/*.json src/abi/**/*.json",
+    "postinstall": "yarn typechain:build && yarn lingui:compile"
```
are you ok with putting `yarn lingui:compile` in a postinstall hook? This would make it run every time a dev types `yarn` or `yarn install`

biwano replied:
Yeah I am okay. Lets talk about this with Jesusf since it has impacts on CI too. He added that:
`
    "prebuild": "git submodule update --init --recursive",
    "postbuild": "yarn lingui:compile",
`
Maybe it is better to have everything in postinstall

appleseed-iii commented on src/views/Stake/Stake.jsx:
```diff
@@ -386,9 +386,6 @@ function Stake() {
                       </Typography>
-                      <Typography variant="body1">
-                        <Trans>Your Balance</Trans>
-                      </Typography>
```
was duped in view

appleseed-iii commented on src/views/Stake/Stake.jsx:
```diff
@@ -398,9 +395,6 @@ function Stake() {
                       </Typography>
-                      <Typography variant="body1">
-                        <Trans>Your Staked Balance</Trans>
-                      </Typography>
```
was duplicated in view

Commit:
Fixed dependencies
Added some translations in Stake.jsx
Fixed useStyle being imported and overridden in src/lib/ohmToast.js

appleseed-iii committed:
adding more translatable strings

appleseed-iii committed:
one more

appleseed-iii commented on src/lib/ohmToast.js:
```diff
@@ -1,4 +1,3 @@
-import { useStyles } from "react";
```
extra import was throwing errors in `lingui:extract`

appleseed-iii committed:
update gitignore for cypress videos

appleseed-iii commented on yarn.lock:
```diff
@@ -1301,6 +1342,16 @@
   resolved "https://registry.yarnpkg.com/@emotion/unitless/-/unitless-0.7.5.tgz#77211291c1900a700b8a78cfafda3160d76949ed"
   integrity sha512-OWORNpfjMsSSUBVrRBVGECkhWcULOAJz9ZW8uK9qgxD+87M7jHRcvh/A96XXNhXTLmKcoYSQtBEX7lHMO7YRwg==
 
+"@endemolshinegroup/cosmiconfig-typescript-loader@^3.0.2":
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -4931,6 +5118,21 @@ checkpoint-store@^1.1.0:
   dependencies:
     functional-red-black-tree "^1.0.1"
 
+chokidar@3.5.1:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -3500,7 +3636,7 @@ anymatch@^2.0.0:
     normalize-path "^2.1.1"
 
-anymatch@^3.0.0, anymatch@^3.0.3, anymatch@~3.1.2:
+anymatch@^3.0.0, anymatch@^3.0.3, anymatch@~3.1.1, anymatch@~3.1.2:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -15969,6 +16454,18 @@ ts-invariant@^0.9.0:
   dependencies:
     tslib "^2.1.0"
 
+ts-node@^9:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -3628,6 +3764,11 @@ are-we-there-yet@~1.1.2:
     delegates "^1.0.0"
     readable-stream "^2.0.6"
 
+arg@^4.1.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -4264,6 +4433,15 @@ batch@0.6.1:
   resolved "https://registry.yarnpkg.com/batch/-/batch-0.6.1.tgz#dc34314f4e679318093fc760272525f94bf25c16"
   integrity sha1-3DQxT05nkxgJP8dgJyUl+UvyXBY=
 
+bcp-47@^1.0.7:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -11983,6 +12389,21 @@ optionator@^0.9.1:
     type-check "^0.4.0"
     word-wrap "^1.2.3"
 
+ora@^5.1.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -4328,6 +4506,15 @@ bip39@^3.0.2:
     pbkdf2 "^3.0.9"
     randombytes "^2.0.1"
 
+bl@^4.1.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -4646,7 +4833,7 @@ buffer@^4.3.0:
     isarray "^1.0.0"
 
-buffer@^5.4.3:
+buffer@^5.4.3, buffer@^5.5.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -8962,6 +9252,25 @@ inquirer@^6.2.2:
     strip-ansi "^5.1.0"
     through "^2.3.6"
 
+inquirer@^7.3.3:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -7753,6 +8028,11 @@ fined@^1.0.1:
     object.pick "^1.2.0"
     parse-filepath "^1.0.1"
 
+flag-icon-css@^3.2.1:
```
flag language switcher

appleseed-iii commented on yarn.lock:
```diff
@@ -5032,6 +5234,18 @@ cli-cursor@^2.1.0:
   dependencies:
     restore-cursor "^2.0.0"
 
+cli-cursor@^3.1.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -5032,6 +5234,18 @@ cli-cursor@^2.1.0:
+  dependencies:
+    restore-cursor "^3.1.0"
+
+cli-spinners@^2.5.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -5042,11 +5256,23 @@ cli-table3@^0.5.1:
   optionalDependencies:
     colors "^1.1.2"
 
+cli-table@^0.3.1:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -5042,11 +5256,23 @@ cli-table3@^0.5.1:
   resolved "https://registry.yarnpkg.com/cli-width/-/cli-width-2.2.1.tgz#b0433d0b4e9c847ef18868a4ef16fd5fc8271c48"
   integrity sha512-GRMWDxpOB6Dgk2E5Uo+3eEBvtOOlimMmpbFiKuLFnQzYDavtLFY3K5ona41jgN/WdRZtG7utuVSVTL4HbZHGkw==
 
+cli-width@^3.0.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -16614,6 +17111,13 @@ wbuf@^1.1.0, wbuf@^1.7.3:
   dependencies:
     minimalistic-assert "^1.0.0"
 
+wcwidth@^1.0.1:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -6038,6 +6294,13 @@ default-resolution@^2.0.0:
   resolved "https://registry.yarnpkg.com/default-resolution/-/default-resolution-2.0.0.tgz#bcb82baa72ad79b426a76732f1a81ad6df26d684"
   integrity sha1-vLgrqnKtebQmp2cy8aga1t8m1oQ=
 
+defaults@^1.0.3:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -5084,6 +5310,11 @@ clone-stats@^1.0.0:
   resolved "https://registry.yarnpkg.com/clone-stats/-/clone-stats-1.0.0.tgz#b3782dff8bb5474e18b9b6bf0fdfe782f8777680"
   integrity sha1-s3gt/4u1R04Yuba/D9/ngvh3doA=
 
+clone@^1.0.2:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -5207,6 +5438,11 @@ colorette@^1.2.1, colorette@^1.2.2, colorette@^1.3.0:
   resolved "https://registry.yarnpkg.com/colorette/-/colorette-1.3.0.tgz#ff45d2f0edb244069d3b772adeb04fed38d0a0af"
   integrity sha512-ecORCqbSFP7Wm8Y6lyqMJjexBQqXSF7SSeaTyGGphogUjBlFP9m9o08wy86HL2uB7fMTxtOUzLMk7ogKcxMg1w==
 
+colors@1.0.3:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -13314,6 +13750,13 @@ prr@~1.0.1:
   resolved "https://registry.yarnpkg.com/prr/-/prr-1.0.1.tgz#d3fc114ba06995a45ec6893f484ceb1d78f5f476"
   integrity sha1-0/wRS6BplaRexok/SEzrHXj19HY=
 
+pseudolocale@^1.1.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -5228,6 +5464,11 @@ command-line-args@^4.0.7:
     find-replace "^1.0.3"
     typical "^2.6.1"
 
+commander@*:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -5501,6 +5747,11 @@ create-hmac@^1.1.0, create-hmac@^1.1.4, create-hmac@^1.1.7:
     safe-buffer "^5.0.1"
     sha.js "^2.4.8"
 
+create-require@^1.1.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -5932,6 +6183,11 @@ data-urls@^2.0.0:
     whatwg-mimetype "^2.3.0"
     whatwg-url "^8.0.0"
 
+date-fns@^2.16.1:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -6153,6 +6416,11 @@ diff-sequences@^27.0.6:
   resolved "https://registry.yarnpkg.com/diff-sequences/-/diff-sequences-27.0.6.tgz#3305cb2e55a033924054695cc66019fd7f8e5723"
   integrity sha512-ag6wfpBFyNXZ0p8pcuIDS//D8H062ZQJ3fzYxjpmeKjnz8W4pekL3AI8VohmyZmsWW2PWaHgjsmqR6L13101VQ==
 
+diff@^4.0.1:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -10909,13 +11285,23 @@ make-dir@^3.0.0, make-dir@^3.0.2:
   dependencies:
     semver "^6.0.0"
 
+make-error@^1, make-error@^1.1.1:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -7606,6 +7874,13 @@ figures@^2.0.0:
   dependencies:
     escape-string-regexp "^1.0.5"
 
+figures@^3.0.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -9159,6 +9468,19 @@ is-accessor-descriptor@^1.0.0:
+  resolved "https://registry.yarnpkg.com/is-alphabetical/-/is-alphabetical-1.0.4.tgz#9e7d6b94916be22153745d184c298cbf986a686d"
+  integrity sha512-DwzsA04LQ10FHTZuL0/grVDk4rFoVH1pjAToYwBrHSxcrBIGQuXrQMtD5U1b0U2XVgKZCTLLP8u2Qxqhy3l2Vg==
+
+is-alphanumerical@^1.0.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -9159,6 +9468,19 @@ is-accessor-descriptor@^1.0.0:
   dependencies:
     kind-of "^6.0.0"
 
+is-alphabetical@^1.0.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -9263,6 +9585,11 @@ is-date-object@^1.0.1:
   dependencies:
     has-tostringtag "^1.0.0"
 
+is-decimal@^1.0.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -9350,6 +9682,13 @@ is-generator-fn@^2.0.0:
   resolved "https://registry.yarnpkg.com/is-generator-fn/-/is-generator-fn-2.1.0.tgz#7d140adc389aaf3011a8f2a2a4cfa6faadffb118"
   integrity sha512-cTIB4yPYL/Grw0EaSzASzg6bBy9gqCofvWN8okThAYIxKJZC+udlRAmGbM0XLeniEJSs8uEgHPGuHSe1XsOLSQ==
 
+is-glob@^2.0.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -9313,6 +9640,11 @@ is-extendable@^1.0.1:
   dependencies:
     is-plain-object "^2.0.4"
 
+is-extglob@^1.0.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -9374,6 +9713,18 @@ is-in-browser@^1.0.2, is-in-browser@^1.1.3:
   resolved "https://registry.yarnpkg.com/is-in-browser/-/is-in-browser-1.1.3.tgz#56ff4db683a078c6082eb95dad7dc62e1d04f835"
   integrity sha1-Vv9NtoOgeMYILrldrX3GLh0E+DU=
 
+is-interactive@^1.0.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -9552,6 +9908,13 @@ is-valid-glob@^1.0.0:
   resolved "https://registry.yarnpkg.com/is-valid-glob/-/is-valid-glob-1.0.0.tgz#29bf3eff701be2d4d315dbacc39bc39fe8f601aa"
   integrity sha1-Kb8+/3Ab4tTTFdusw5vDn+j2Aao=
 
+is-valid-path@^0.1.1:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -9374,6 +9713,18 @@ is-in-browser@^1.0.2, is-in-browser@^1.1.3:
+  resolved "https://registry.yarnpkg.com/is-interactive/-/is-interactive-1.0.0.tgz#cea6e6ae5c870a7b0a0004070b7b587e0252912e"
+  integrity sha512-2HvIEKRoqS62guEC+qBjpvRubdX910WCMuJTZ+I9yvqKU2/12eSL549HMwtabb4oupdj2sMP50k+XJfB/8JE6w==
+
+is-invalid-path@^0.1.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -10825,11 +11193,19 @@ lodash.uniq@^4.5.0:
   resolved "https://registry.yarnpkg.com/lodash/-/lodash-4.17.21.tgz#679591c564c3bffaae8454cf0b3df370c3d6911c"
   integrity sha512-v2kDEe57lecTulaDIuNTPy3Ry4gLGJ6Z1O3vE1krgXZNrsQ+LFTGHVxVjcXPs17LhbZVGedAJv8XZ1tvj5FvSg==
 
+log-symbols@^4.1.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -9542,6 +9893,11 @@ is-unc-path@^1.0.0:
   dependencies:
     unc-path-regex "^0.1.2"
 
+is-unicode-supported@^0.1.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -10785,6 +11148,11 @@ lodash.defaults@^4.2.0:
   resolved "https://registry.yarnpkg.com/lodash.defaults/-/lodash.defaults-4.2.0.tgz#d09178716ffea4dde9e5fb7b37f6f0802274580c"
   integrity sha1-0JF4cW/+pN3p5ft7N/bwgCJ0WAw=
 
+lodash.get@^4, lodash.get@^4.4.2:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -10909,13 +11285,23 @@ make-dir@^3.0.0, make-dir@^3.0.2:
   dependencies:
     kind-of "^6.0.2"
 
+make-plural@^6.2.2:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -11107,6 +11493,11 @@ meros@^1.1.4:
   resolved "https://registry.yarnpkg.com/meros/-/meros-1.1.4.tgz#c17994d3133db8b23807f62bec7f0cb276cfd948"
   integrity sha512-E9ZXfK9iQfG9s73ars9qvvvbSIkJZF5yOo9j4tcwM5tN8mUKfj/EKN5PzOr3ZH0y5wL7dLAHw3RVEfpQV9Q7VQ==
 
+messageformat-parser@^4.1.3:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -11117,6 +11508,14 @@ microevent.ts@~0.1.1:
   resolved "https://registry.yarnpkg.com/microevent.ts/-/microevent.ts-0.1.1.tgz#70b09b83f43df5172d0205a63025bce0f7357fa0"
   integrity sha512-jo1OfR4TaEwd5HOrt5+tAZ9mqT4jmpNAusXtyfNzqVm9uiSYFZlKM1wYL4oU7azZW/PxQW53wM0S6OR1JHNa2g==
 
+micromatch@4.0.2:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -11529,6 +11928,13 @@ node-forge@^0.10.0:
   resolved "https://registry.yarnpkg.com/node-forge/-/node-forge-0.10.0.tgz#32dea2afb3e9926f02ee5ce8794902691a676bf3"
   integrity sha512-PPmu8eEeG9saEUvI97fm4OYxXVB6bFvyNTyiUOBichBpFG8A1Ljw3bY62+5oOjDEMHRnd0Y7HQ+x7uzxOzC6JA==
 
+node-gettext@^3.0.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -12118,6 +12539,11 @@ pako@~1.0.5:
   resolved "https://registry.yarnpkg.com/pako/-/pako-1.0.11.tgz#6c9599d340d54dfd3946380252a35705a6b992bf"
   integrity sha512-4hLB8Py4zZce5s4yd9XzopqwVv/yGNhV1Bl8NTmCq1763HeK2+EwVTv+leGeL13Dnh2wfbqowVPXCIO0z4taYw==
 
+papaparse@^5.3.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -12434,6 +12860,11 @@ plur@^3.0.0:
   dependencies:
     irregular-plurals "^2.0.0"
 
+plurals-cldr@^1.0.4:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -12446,6 +12877,11 @@ pnp-webpack-plugin@1.6.4:
   dependencies:
     ts-pnp "^1.1.6"
 
+pofile@^1.1.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -13477,6 +13920,11 @@ raf@^3.4.0, raf@^3.4.1:
   dependencies:
     performance-now "^2.1.0"
 
+ramda@^0.27.1:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -13594,6 +14042,15 @@ react-event-listener@^0.6.2:
     prop-types "^15.6.0"
     warning "^4.0.1"
 
+react-flag-icon-css@^1.0.25:
```
language switcher flag

appleseed-iii commented on yarn.lock:
```diff
@@ -13934,6 +14391,13 @@ readdirp@^2.2.1:
     micromatch "^3.1.10"
     readable-stream "^2.0.2"
 
+readdirp@~3.5.0:
```
react-flag

appleseed-iii commented on yarn.lock:
```diff
@@ -14230,6 +14694,11 @@ require-main-filename@^2.0.0:
   resolved "https://registry.yarnpkg.com/require-main-filename/-/require-main-filename-2.0.0.tgz#d0b329ecc7cc0f61649f62215be69af54aa8989b"
   integrity sha512-NKN5kMDylKuldxYLSUfrbo5Tuzh4hd+2E8NPPX02mZtn1VuREQToYe/ZdlJy+J3uCpfaiGF05e7B8W0iXbQHmg==
 
+require-package-name@^2.0.1:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -14352,6 +14821,14 @@ restore-cursor@^2.0.0:
     onetime "^2.0.0"
     signal-exit "^3.0.2"
 
+restore-cursor@^3.1.0:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -15017,6 +15494,14 @@ source-map-resolve@^0.6.0:
     atob "^2.1.2"
     decode-uri-component "^0.2.0"
 
+source-map-support@^0.5.17:
```
lingui dependency

appleseed-iii commented on yarn.lock:
```diff
@@ -17292,6 +17796,11 @@ yargs@^7.1.0:
     y18n "^3.2.1"
     yargs-parser "^5.0.1"
 
+yn@3.1.1:
```
lingui dependency

appleseed-iii reviewed (approved):
ready to rock.

Zayen-X commented:
YEEE
